apps/builder/app/builder/features/sidebar-left/panels/pages/settings.tsx

```diff
@@ -255,20 +235,43 @@ export const NewPageSettings = ({
 
   const handleSubmit = () => {
     if (Object.keys(errors).length === 0) {
-      fetcher.submit(values, {
-        method: "put",
-        action: restPagesPath({ projectId }),
-      });
+      const pageId = nanoid();
+      store.createTransaction(
+        [pagesStore, instancesStore],
+        (pages, instances) => {
+          if (pages === undefined) {
+            return;
```

Comment:
when this happens, can we still call onSuccess()?

Reply:
Never. Just type guard. It will be removed we replace current pages format to map like all other data.